### PR TITLE
Added Popover, Tooltip, and Tabs

### PR DESCRIPTION
**Pull request by depiction**
This adds the following components to Robits from the Unshuffled project: Popover, Tooltip, and Tabs. I converted the Popover and Tooltip components from class components to functional components. The Tabs component was already a functional component.

I've noticed that Storybook doesn't always load components properly. I sometimes have to switch from the canvas to docs tab, then back to the canvas tab for the component to load and work properly.

### CODE REVIEW

File: src/core/components/popover/popover_talentPortal.module.scss [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+.popover {
+  position: absolute;
+  top: 0;
+  left: 0;
+  z-index: $zindex-popover;
+  display: block;
+  max-width: $popover-max-width;
+  // Our parent element can be arbitrary since tooltips are by default inserted as a sibling of their target element.
+  // So reset our font and text properties to avoid inheriting weird values.
+  // @include reset-text();
```
Comment (dfagan2) [inline src/core/components/popover/popover_talentPortal.module.scss:39]: i know this is copy-paste and leftover, but we can remove it

File: src/core/components/popover/Popover.js [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+  toggle,
+  styling,
+  ...attrs
+}) => {
+  const popoverContentAttrs = omit(attrs, ['disabled', 'delay', 'toggle'])
+  const [_target, _setTarget] = useState(null)
+  let _hideTimeout = null
+  let _showTimeout = null
+
+  const classes = classNames(innerClassName)
```
Comment (dfagan2) [inline src/core/components/popover/Popover.js:40]: is this translation through `classNames` needed? could `innerClassName` be applied directly?

File: src/core/components/tabs/Tabs.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
+import React, { useEffect, useState } from 'react'
+import classNames from 'classnames'
+import { KEYCODES } from '../../constants/constants'
+import ThemeWrapper from '../../utils/ThemeWrapper'
+
+export const Tabs = ({
+  options = [],
```
Comment (dfagan2) [inline src/core/components/tabs/Tabs.js:7]: if I'm remembering correctly, I believe in order for the docs tab to successfully pull in this information, it has to be pulled out into a dedicated `Tabs.propTypes = {}` definition at the bottom of the file like we are doing elsewhere. Hopefully that change fixes the missing info and "unknowns" here:
Screen Shot 2021-02-25 at 1 14 41 PM

File: src/core/components/tooltip/Tooltip.js [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+  let _hideTimeout = null
+  let _showTimeout = null
+
+  useEffect(() => {
+    _setTarget(layoutUtils.getTarget(target))
+
+    return () => {
+      clearTimeout(_showTimeout)
+      clearTimeout(_hideTimeout)
+      removeListeners()
```
Comment (dfagan2) [inline src/core/components/tooltip/Tooltip.js:46]: This unmount is pretty consistently throwing the below error. Like when you navigate away from the Tooltips story, a second or two later the new story you landed on errors. Could add some optional chaining there to help prevent that error, but may be a better way?
Screen Shot 2021-02-25 at 1 13 31 PM

File: src/core/components/tooltip/Tooltip.js [hunk 5]
```diff
@@ -0,0 +1,365 @@
...
+    }
+
+    return toggle(openRef.current, e)
+  }
+
+  if (!open) {
+    return null
+  }
+
+  const classes = classNames(styling['tooltip-inner'], innerClassName)
```
Comment (dfagan2) [inline src/core/components/tooltip/Tooltip.js:217]: `innerContentClasses` would be a better name here

File: src/core/components/tooltip/tooltip_talentPortal.module.scss [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+$tooltip-arrow-width: 5px !default;
+$tooltip-arrow-height: 5px !default;
+$tooltip-arrow-color: $tooltip-bg !default;
+
+// Base class
+.tooltip {
+  position: absolute;
+  display: block;
+  // Our parent element can be arbitrary since tooltips are by default inserted as a sibling of their target element.
+  // So reset our font and text properties to avoid inheriting weird values.
```
Comment (dfagan2) [inline src/core/components/tooltip/tooltip_talentPortal.module.scss:24]: the above 2 comment lines can be deleted

**Review by dfagan2 — CHANGES_REQUESTED** [review]
Looking great, nice work. Just a couple things..

**Review by dfagan2 — APPROVED** [review]
There are some unused imports in `Tabs.stories.js`, then we can merge

**Comment by depiction** [comment]
@dfagan2 I removed the unused imports. Github won't let me merge this until you approve it again.

**Review by dfagan2 — APPROVED** [review]

**Pull request merged**